scripts/quick-helm-install.sh

```diff
@@ -0,0 +1,111 @@
+        echo -e "Usage:\n\tquick-helm-install.sh [options]"
+        echo " "
+        echo "Installs StackRox via Helm charts, requires Helm."
+        echo " "
+        echo "options:"
+        echo "-h, --help                show brief help"
+        echo "-s, --small               reduce StackRox resource requirements for small clusters"
+        exit 0
+        ;;
+    -s|--small)
+        SMALL_INSTALL=true
+        ;;
+esac
+
+echo "Adding the stackrox/helm-charts/opensource repository to Helm."
+
+helm repo add stackrox https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource/
+
+echo "Generating STACKROX_ADMIN_PASSWORD"
+
+STACKROX_ADMIN_PASSWORD="$(openssl rand -base64 20 | tr -d '/=+')"
+
+echo "Installing stackrox-central-services"
+
+helm install -n stackrox --create-namespace stackrox-central-services stackrox/stackrox-central-services --set central.adminPassword.value="${STACKROX_ADMIN_PASSWORD}"
+
+if $SMALL_INSTALL; then
+    echo "Upgrading stackrox to a small install"
+    helm upgrade -n stackrox stackrox-central-services stackrox/stackrox-central-services \
+      --set central.resources.requests.memory=1Gi \
+      --set central.resources.requests.cpu=1 \
+      --set central.resources.limits.memory=4Gi \
+      --set central.resources.limits.cpu=1 \
+      --set scanner.autoscaling.disable=true \
+      --set scanner.replicas=1 \
+      --set scanner.resources.requests.memory=500Mi \
+      --set scanner.resources.requests.cpu=500m \
+      --set scanner.resources.limits.memory=2500Mi \
+      --set scanner.resources.limits.cpu=2000m
+fi
+
+kubectl -n stackrox rollout status deploy/central --timeout=3m
+
+echo "Forwarding ports to central"
+
+kubectl -n stackrox port-forward deploy/central --pod-running-timeout=1m0s 8000:8443 > /dev/null 2>&1 &
+
+echo "Generating an init bundle with shared secrets"
+
+kubectl -n stackrox exec deploy/central -- roxctl --insecure-skip-tls-verify \
+  --password "${STACKROX_ADMIN_PASSWORD}" \
+  central init-bundles generate stackrox-init-bundle --output - 1> stackrox-init-bundle.yaml
+
+installflags=("-f stackrox-init-bundle.yaml" "--set clusterName=\"my-secured-cluster\"")
+if $SMALL_INSTALL; then
+    installflags+=("--set sensor.resources.requests.memory=500Mi")
+    installflags+=("--set sensor.resources.requests.cpu=500m")
+    installflags+=("--set sensor.resources.limits.memory=500Mi")
+    installflags+=("--set sensor.resources.limits.cpu=500m")
+    echo "Installing the first small secured cluster"
+else
+    echo "Installing the first secured cluster"
+fi
+
+helm install -n stackrox stackrox-secured-cluster-services stackrox/stackrox-secured-cluster-services ${installflags[*]}
+
+echo "Logging into StackRox in the browser"
+
+curl_status=0
+target_url="$(curl -sSkf -u "admin:${STACKROX_ADMIN_PASSWORD}" -o /dev/null -w '%{redirect_url}' "https://localhost:8000/sso/providers/basic/4df1b98c-24ed-4073-a9ad-356aec6bb62d/challenge?micro_ts=0")" || curl_status=$?
```

Comment:
This is fishy - I can't believe `shellcheck` doesn't complain here :) 

I believe we require `|| curl_status=$?` because of running with `-e`. 
This code uses `curl_status` only once, so maybe we could turn it into something like:

```shell
if curl -sSkf -u "admin:${STACKROX_ADMIN_PASSWORD}" -o /dev/null -w '%{redirect_url}' "https://localhost:8000/sso/providers/basic/4df1b98c-24ed-4073-a9ad-356aec6bb62d/challenge?micro_ts=0"; then
    if [[ -n "$target_url" ]]; then
    ...
    fi
else
    echo "Could not determine login URL for basic auth user" 1>&2
fi
```

The code also checks `-n "$target_url"` twice and I think once is enough.

Reply:
I don't see where target_url is ever set in your example. The curl line is taken like this from the logmein.sh script and it works there 😛

Reply:
Sorry, I am hesitant to buy an argument "it works over there" 😂  Tricky question for you to exercise: having a command `a || b` - do you know when `b` is be executed and when not?

I see my solution does not store the value that you need - my bad I missed it. Have you maybe tried to replace `-o /dev/null` with something else?

Would something like this work better?

```shell
local target_url
local _output
_output="$(mktemp)"
if curl -sSkf -u "admin:${STACKROX_ADMIN_PASSWORD}" --output "${_output}" -w '%{redirect_url}' "https://localhost:8000/sso/providers/basic/4df1b98c-24ed-4073-a9ad-356aec6bb62d/challenge?micro_ts=0"; then
    target_url="$(cat "${_output}")"
    if [[ -n "$target_url" ]]; then
    ...
    fi
else
    echo "Could not determine login URL for basic auth user" 1>&2
fi
```

Reply:
Yeah this works better, thanks. I'm not fully familiar with the deeper intricacies of bash yet so the question regarding `a || b` isn't immediately obvious to me. My assumption would be that if a is false b doesn't get executed in order to be efficient, but I'd have to look it up.

Reply:
This was only a thought-exercise, so all good :) Answer: if `a` is true, `b` will not be executed.